pkg/daemon/update.go

```diff
@@ -1484,6 +1484,7 @@ func (dn *Daemon) updateOS(config *mcfgv1.MachineConfig, osImageContentDir strin
 	glog.Infof("Updating OS to %s", newURL)
 	client := NewNodeUpdaterClient()
 	if _, err := client.Rebase(newURL, osImageContentDir); err != nil {
+		MCDPivotErr.WithLabelValues(dn.node.Name, newURL, err.Error()).SetToCurrentTime()
```

Comment:
updateOS() unit test segfaults if it reaches here. Made unsuccessful attempt to start a fake metrics listener.

Reply:
since this is the only place returning an error we could just move this to capture the return value of updateOS and leave your unit tests alone?

like here: https://github.com/openshift/machine-config-operator/blob/dd9050c7faf5400cde55608a0393f90bfc213e53/pkg/daemon/update.go#L350

Reply:
yes, we can. There is one more place which calls updateOS() https://github.com/openshift/machine-config-operator/blob/master/pkg/daemon/daemon.go#L1036 but I really don't know if it is ever called. I updated this code to maintain the compatibility with previous code.

Reply:
moved it on level up